### PR TITLE
updates to "align_multiple_visits" and "align_sparse_fields" notebooks

### PR DESCRIPTION
**Pull request by gsanand**
This notebook checklist has been made available to us by the [Notebooks For All team](https://github.com/Iota-School/notebooks-for-all/blob/main/resources/event-hackathon/notebook-authoring-checklist.md).
Its purpose is to serve as a guide for both the notebook author and the technical reviewer highlighting critical aspects to consider when striving to develop an accessible and effective notebook.

### The First Cell

- [ ] The title of the notebook in a first-level heading (eg. `<h1>` or `# in markdown`).
- [ ] A brief description of the notebook.
- [ ] A table of contents in an [ordered list](https://www.markdownguide.org/basic-syntax/#ordered-lists) (`1., 2.,` etc. in Markdown).
- [ ] The author(s) and affiliation(s) (if relevant).
- [ ] The date first published.
- [ ] The date last edited (if relevant).
- [ ] A link to the notebook's source(s) (if relevant).

### The Rest of the Cells

- [ ] There is only one H1 (`#` in Markdown) used in the notebook.
- [ ] The notebook uses other [heading tags](https://www.markdownguide.org/basic-syntax/#headings) in order (meaning it does not skip numbers). 

## Text

- [ ] All link text is descriptive. It tells users where they will be taken if they open the link.
- [ ] All acronyms are defined at least the first time they are used. 
- [ ] Field-specific/specialized terms are used when needed, but not excessively.

## Code

- [ ] Code sections are introduced and explained before they appear in the notebook. This can be fulfilled with a heading in a prior Markdown cell, a sentence preceding it, or a code comment in the code section.
- [ ] Code has explanatory comments (if relevant). This is most important for long sections of code.
- [ ] If the author has control over the syntax highlighting theme in the notebook, that theme has enough color contrast to be legible.
- [ ] Code and code explanations focus on one task at a time. Unless comparison is the point of the notebook, only one method for completing the task is described at a time.

### Images

- [ ] All images (jpg, png, svgs) have an [image description](https://www.w3.org/WAI/tutorials/images/decision-tree/). This could be
    - [ ] [Alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt) (an `alt` property) 
    - [ ] [Empty alt text](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/alt#decorative_images) for decorative images/images meant to be skipped (an `alt` attribute with no value)
    - [ ] Captions
    - [ ] If no other options will work, the image is decribed in surrounding paragraphs.

- [ ] Any [text present in images](https://www.w3.org/WAI/WCAG21/Understanding/images-of-text.html) exists in a text form outside of the image (this can be alt text, captions, or surrounding text.)

### Visualizations

- [ ] All visualizations have an image description. Review the previous section, Images, for more information on how to add it.
- [ ] [Visualization descriptions](http://diagramcenter.org/specific-guidelines-e.html) include
    - [ ] The type of visualization (like bar chart, scatter plot, etc.)
    - [ ] Title
    - [ ] Axis labels and range
    - [ ] Key or legend
    - [ ] An explanation of the visualization's significance to the notebook (like the trend, an outlier in the data, what the author learned from it, etc.)

- [ ] All visualizations and their parts have [enough color contrast](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html) ([color contrast checker](https://webaim.org/resources/contrastchecker/)) to be legible. Remember that transparent colors have lower contrast than their opaque versions.
- [ ] All visualizations [convey information with more visual cues than color coding](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html). Use text labels, patterns, or icons alongside color to achieve this.
- [ ] All visualizations have an additional way for notebook readers to access the information. Linking to the original data, including a table of the data in the same notebook, or sonifying the plot are all options.

### CODE REVIEW

**Comment by mackjenn** [comment]
Would it be more straightforward for CI testing to have these two notebooks in separate PRs?

**Comment by gsanand** [comment]
@mackjenn good point, I'm not sure. @haticekaratay what would you prefer? happy to make the switch if desired.

File: notebooks/DrizzlePac/align_multiple_visits/align_multiple_visits.ipynb [hunk 1]
```diff
@@ -4,31 +4,75 @@
    "cell_type": "markdown",
```
Comment (haticekaratay) [inline notebooks/DrizzlePac/align_multiple_visits/align_multiple_visits.ipynb:4]: <p>Hi @gsanand, the files <code>residuals_{rootname_A}_flc.png"</code>  and  <code>residuals_{rootname_B}_flc.png",</code></p><p>have already been renamed previosly with a trailing<code>_rscale</code> which is why they don't exist. Perhaps, you'll need to change the workflow on this. Thanks</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/hst_notebooks/pull/266/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (gsanand) [reply notebooks/DrizzlePac/align_multiple_visits/align_multiple_visits.ipynb:4]: Hi @haticekaratay , the errors I see in the CI are with regards to opening the "_rscale" or other similar files though, and not the ones that end in "_flc" themselves.

for example:

FileNotFoundError: [Errno 2] No such file or directory: 'residuals_j9irw5kaq_flc_rscale.png'

Comment (haticekaratay) [reply notebooks/DrizzlePac/align_multiple_visits/align_multiple_visits.ipynb:4]: Yes, indeed. You are right; I was looking at the previous build logs. Let's get @dulude and @mgough-970's feedback. Hi @dulude and @mgough-970, this notebook (align_multiple_visits) runs successfully locally and produces no errors with local tests:
 `pytest --nbval ./notebooks/DrizzlePac/align_multiple_visits/align_multiple_visits.ipynb`
  and ` jupyter nbconvert --to html --execute ./notebooks/DrizzlePac/align_multiple_visits/align_multiple_visits.ipynb`. However fails on the validation step in CI.

**Comment by gsanand** [comment]
@mgough-970 @haticekaratay  I tried to modify the file handling in hopes that it would help the CI, but now the CI can't even find the image file that is made as part of the notebook itself. I don't really know what to do from here

**Comment by mackjenn** [comment]
Can we try commenting out any cells that display PNG files to see if the notebook will pass?

**Comment by mackjenn** [comment]
One idea is that it could be a permissions issue, so we could try doing 'chmod a+rwx *png'.

**Comment by gsanand** [comment]
@mrevalski how does the pinning of photutils=1.12 work, if I'm not calling it explicitly? (its being called through drizzlepac). 

@haticekaratay at some point the align_sparse notebook was moved to a different pull request, but it looks like its back now?

File: notebooks/DrizzlePac/align_multiple_visits/requirements.txt
Comment (mgough-970) [inline notebooks/DrizzlePac/align_multiple_visits/requirements.txt:1]: This is structured as an environmental definition file, not a requirements file.

Comment (mgough-970) [reply notebooks/DrizzlePac/align_multiple_visits/requirements.txt:1]: python should also be 3.11 to match all other repo requirements

Comment (gsanand) [reply notebooks/DrizzlePac/align_multiple_visits/requirements.txt:1]: @mgough-970 yes, I was going off some recommendations from Megan Sosey but did not have time to fully implement them yesterday. the file was previously a proper requirements.txt file– the CI was not passing all checks at that time either

Comment (gsanand) [reply notebooks/DrizzlePac/align_multiple_visits/requirements.txt:1]: reverted to a prior version now.

**Comment by gsanand** [comment]
hi @haticekaratay, we (INS) are trying to get this work closed out ASAP. can you check what the new "bandit" error messages are about? also, can we remove align_sparse_fields from this pull request, since that is already passed in another pull request?

thank you!

**Comment by gsanand** [comment]
also now there is an issue where it complains the "image" directory already exists, when it doesn't until its made as part of the relevant cell?  I'm at a total loss at this point, the CI errors do not make sense and are effectively impossible for me to debug.

**Pull request closed without merging**